src/app/auth/services/auth.service.ts

```diff
@@ -0,0 +1,52 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ILoginRequestDto } from '../dto/loginRequest.dto';
+import { ILoginResponseDto } from '../dto/loginResponse.dto';
+import { IRegisterRequestDto } from '../dto/registerRequest.dto';
+import { IRegisterResponseDto } from '../dto/registerResponse.dto';
+import { SessionStorageService } from './session-storage.service';
+
+import { environment } from '../../../environments/environment';
+
+@Injectable({ providedIn: 'root' })
+export class AuthService {
+  private _isAuthorized$$!: BehaviorSubject<boolean>;
+  public isAuthorized$!: Observable<boolean>;
+
+  constructor(
+    private http: HttpClient,
+    private sessionStorageService: SessionStorageService,
+    private router: Router
+  ) {
+    const token = this.sessionStorageService.getToken();
+    this._isAuthorized$$ = new BehaviorSubject<boolean>(!!token);
```

Comment:
Please initialize it on line 16, it will be cleaner and will not affect logic an any way